Dockerfile

```diff
@@ -3,64 +3,62 @@ FROM node:10-alpine AS build
 ARG NPM_TOKEN
 
 # Make Git available for NPM and rsync in the build image
-RUN apk add --update git rsync && \
-  rm -rf /tmp/* /var/cache/apk/*
+RUN apk add --update git && rm -rf /var/cache/apk/*
 
-COPY . /tmp/
+WORKDIR /build
+COPY . .
 
 # Only if needed, copy .npmrc files into the container
-# COPY Dockerfile.npmrc /tmp/.npmrc
-# COPY .npmrc /tmp/.npmrc
+# COPY Dockerfile.npmrc /build/.npmrc
+# COPY .npmrc /build/.npmrc
 
-RUN cd /tmp && npm install --production --verbose
-RUN rsync -azhqi /tmp/node_modules/ /tmp/production_node_modules
+RUN npm install --production --verbose && mv node_modules production_node_modules
 
 # Dev dependencies
-RUN cd /tmp && npm install --verbose
-RUN rm -rf /tmp/.npmrc
+RUN npm install --verbose && rm -rf .npmrc
 
 # TypeScript build
-RUN cd /tmp && npm run-script build
+RUN npm run-script build
+
+RUN cd default-assets-package && npm install && npm run build
 
 FROM node:10-alpine AS run
-ENV APPDIR=/usr/src/repos
 
-RUN mkdir -p "${APPDIR}"
+ENV IS_DOCKER=1 \
+    DEBUG=oss-initialize \
+    NPM_CONFIG_LOGLEVEL=warn \
+    PORT=3000
 
-# Production Node.js modules
-COPY --from=build /tmp/production_node_modules "${APPDIR}/node_modules"
-
-# Assets that people not using painless config may need
-COPY --from=build /tmp/data "${APPDIR}/data"
-
-# Copy built assets, app, config map
-COPY --from=build /tmp/dist "${APPDIR}"
-COPY --from=build /tmp/config "${APPDIR}/config"
-COPY --from=build /tmp/views "${APPDIR}/views"
-COPY --from=build /tmp/package.json "${APPDIR}/package.json"
-COPY --from=build /tmp/jobs/reports/exemptRepositories.json "${APPDIR}/jobs/reports/"
-COPY --from=build /tmp/jobs/reports/organizationDefinitions.json "${APPDIR}/jobs/reports/"
-COPY --from=build /tmp/jobs/reports/repositoryDefinitions.json "${APPDIR}/jobs/reports/"
-COPY --from=build /tmp/jobs/reports/teamDefinitions.json "${APPDIR}/jobs/reports/"
-COPY --from=build /tmp/jobs/reports/views "${APPDIR}/jobs/reports/views"
+EXPOSE ${PORT}
 
 WORKDIR /usr/src/repos
 
-# COPY package.json "${APPDIR}"
-# COPY views "${APPDIR}/views"
-# COPY dist "${APPDIR}"
-# COPY public "${APPDIR}/public"
+RUN addgroup oss && adduser -D -G oss oss && chown -R oss:oss .
 
-ENV IS_DOCKER=1
-ENV DEBUG=oss-initialize
+# Production Node.js modules
+COPY --from=build --chown=oss:oss /build/production_node_modules ./node_modules
 
-ENV NPM_CONFIG_LOGLEVEL=warn
+# Assets that people not using painless config may need
+COPY --from=build --chown=oss:oss /build/data ./data
 
-ENV PORT 3000
-EXPOSE 3000
+# Copy built assets, app, config map
+COPY --from=build --chown=oss:oss /build/dist ./
```

Comment:
Moved things most likely to change to the bottom so other layes can benefit from cache.

This helps both the build and pull of images.